test/unit/org/apache/cassandra/cql3/validation/operations/CreateTest.java

```diff
@@ -569,6 +575,137 @@ public void testDoubleWith() throws Throwable
             assertInvalidSyntaxMessage("no viable alternative at input 'WITH'", stmt);
     }
 
+    public static class InvalidMemtableFactoryMethod
+    {
+        public static String factory(Map<String, String> options)
+        {
+            return "invalid";
+        }
+    }
+
+    public static class InvalidMemtableFactoryField
+    {
+        public static String FACTORY = "invalid";
+    }
+
+    @Test
+    public void testCreateTableWithMemtable() throws Throwable
+    {
+        createTable("CREATE TABLE %s (a text, b int, c int, primary key (a, b))");
+        assertSame(MemtableParams.DEFAULT.factory, getCurrentColumnFamilyStore().metadata().params.memtable.factory);
+
+        assertRows(execute(format("SELECT memtable FROM %s.%s WHERE keyspace_name = ? and table_name = ?;",
+                                  SchemaConstants.SCHEMA_KEYSPACE_NAME,
+                                  SchemaKeyspaceTables.TABLES),
+                           KEYSPACE,
+                           currentTable()),
+                   row(map()));
```

Comment:
This block appears seven times across the test, only changing the map arguments. We could reduce duplication with an utility method like:
```java
private void assertMemtableOptions(Object... options) throws Throwable
{
    assertRows(execute(format("SELECT memtable FROM %s.%s WHERE keyspace_name = ? and table_name = ?",
                              SchemaConstants.SCHEMA_KEYSPACE_NAME,
                              SchemaKeyspaceTables.TABLES),
                       KEYSPACE,
                       currentTable()),
               row(map(options)));
}
```

Reply:
Done